src/AbstractCsv.php

```diff
@@ -127,6 +128,35 @@ public static function createFromFileObject(SplFileObject $file)
     }
 
     /**
+     * Return a new {@link AbstractCsv} from a PHP resource stream
+     *
+     * @param resource $stream
+     *
+     * @return static
+     */
+    public static function createFromStream($stream)
+    {
+        if (is_resource($stream)) {
+            $stream = new StreamIterator($stream);
+        }
+
+        if ($stream instanceof StreamIterator) {
```

Comment:
I'd suggest to negate this check and throw this block, keeping the happy path alive.